jupyter-images/fall-2024/tm/Dockerfile

```diff
@@ -1,7 +1,7 @@
 # Heavily borrowed from docker-stacks/minimal-notebook/
 # https://github.com/jupyter/docker-stacks/blob/main/minimal-notebook/Dockerfile
 
-ARG BASE_CONTAINER=jupyter/minimal-notebook
+ARG BASE_CONTAINER=jupyter/tensorflow-notebook
```

Comment:
Why the switch from minimal to the tf image?

Reply:
What are you trying? The image that's live right now uses the minimal-notebook and thomas and I can access the GPU just fine.

Reply:
How did you get that to work beyond the usual, i.e., the newly defined `environment.yml` and `jupyterhub_gpu.yaml`. Did you have to install anything special CUDA/GPU-wise?

Reply:
My understanding is that the current running environment was modified "in-place" for faster experimentation so not everything may have been captured in the `Dockerfile`, `environment.yml`, etc. I guess that is what I am asking.

Reply:
You can find the most recent Dockerfile/environment combo in the "ana" tmux session on the docker-gpu machine. Your most recent commit has the correct environment file, however using the minimal-notebook as the base still works.

To get the GPU to work there is nothing more needed past the "normal" things--i.e. requesting the additional resource via `jupyterhub_gpu.yaml`.

I've found that the important parts of getting these things to work are making sure the packages you install (via `conda` or `pip`) are compatible with that shown when doing an `nvidia-smi` on JS2.

For example, in this case I tell `pip` to look for `torch` and associated packages from the cuda 12.1 index with [this line](https://github.com/julienchastang/science-gateway/blob/7364b7959863c8215d327aa7bc51bfebcacb9222/jupyter-images/fall-2024/tm/environment.yml#L29):
`    - --extra-index-url https://download.pytorch.org/whl/cu121`

We can get together to discuss this in-person after the holidays?